### PR TITLE
Fix various Spring DI processor bugs relating to scopes

### PR DESCRIPTION
**Pull request by geoand**
Unblocks: #2179

### CODE REVIEW

**Comment by geoand** [comment]
Failure seems totally unrelated (Opentracing). I will restart it when the other tests are done.

**Comment by geoand** [comment]
@mkouba tests are passing :)

File: extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDIProcessor.java [hunk 1]
```diff
@@ -265,6 +265,14 @@ private void visitAnnotation(final DotName clazz, final IndexView index, final M
             final Set<DotName> scopeStereotypes = new HashSet<>();
             final Set<String> names = new HashSet<>();
             final Set<DotName> clazzAnnotations = classInfo.annotations().keySet();
+
+            for (AnnotationInstance instance : classInfo.classAnnotations()) {
+                // make sure that we don't mix and match Spring and CDI annotations since this can cause a lot of problems
+                if (BuiltinScope.from(instance.name()) != null) {
```
Comment (mkouba) [inline extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDIProcessor.java]: Hm, the only problem here is that the set of scopes is extensible. But I don't think that's something we can fix with the current API. I've created https://github.com/quarkusio/quarkus/issues/2188.

**Review by mkouba — APPROVED** [review]

File: extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDIProcessor.java (hunk 1 above)
Comment (geoand) [reply extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDIProcessor.java]: :+1:

**Pull request merged**